sql/log_event_server.cc

```diff
@@ -6203,6 +6223,7 @@ int Rows_log_event::do_apply_event(rpl_group_info *rgi)
      */
     rgi->set_row_stmt_start_timestamp();
 
+    make_thd_info(thd_info_cache);
```

Comment:
Can the new `Rows_log_event::make_thd_info()` be a part of `Rows_log_event::do_before_row_operations()`?
Maybe, if `thd->variables.sql_mode= MODE_NO_AUTO_VALUE_ON_ZERO;` won’t (or is not meant to) change things.

This function does not (immediately) process the errno return of `do_before_row_operations()`.
This may be a concern.